tests/test_downloader_handlers.py

```diff
@@ -297,27 +312,30 @@ def test_download(self):
     @defer.inlineCallbacks
     def test_download_gzip_response(self):
 
-        if six.PY2 and twisted_version > (12, 3, 0):
+        if twisted_version > (12, 3, 0):
 
             crawler = get_crawler(SingleRequestSpider)
-            body = '1'*100 # PayloadResource requires body length to be 100
+            body = b'1'*100 # PayloadResource requires body length to be 100
             request = Request('http://localhost:8998/payload', method='POST', body=body, meta={'download_maxsize': 50})
             yield crawler.crawl(seed=request)
             failure = crawler.spider.meta['failure']
             # download_maxsize < 100, hence the CancelledError
             self.assertIsInstance(failure.value, defer.CancelledError)
 
-            request.headers.setdefault('Accept-Encoding', 'gzip,deflate')
+            request.headers.setdefault(b'Accept-Encoding', b'gzip,deflate')
             request = request.replace(url='http://localhost:8998/xpayload')
             yield crawler.crawl(seed=request)
 
-            # download_maxsize = 50 is enough for the gzipped response
-            failure = crawler.spider.meta.get('failure')
-            self.assertTrue(failure == None)
-            reason = crawler.spider.meta['close_reason']
-            self.assertTrue(reason, 'finished')
+            if six.PY2:
+                # download_maxsize = 50 is enough for the gzipped response
+                # See issue https://twistedmatrix.com/trac/ticket/8175
+                raise unittest.SkipTest("xpayload only enabled for PY2")
```

Comment:
I messed-up indentation, sorry - pushed a fix. A part of the test is skipped on py3, I think I should be able to unskip it in #1680, where compression is ported.